### ISSUE TITLE
[Bug] GridLength of 0 no longer hides/clips grid row content

### ISSUE THREAD

**Issue by markdou812** [xamarin/Xamarin.Forms#8131]
### Description
Setting the height of a grid row to an absolute size of 0 no longer properly hides/clips the composite control content in that row.
I validated that this is a bug in XF 4.2.0.848062 and is still broken in 4.3.0.851321-pre3.
This used to work in XF 3.4.0.1008975.

### Steps to Reproduce

1. Create a composite control (e.g. containing a label and an image)
2. Add the composite control to a grid row
3. Programmatically set the height of the grid row to 0 absolute.

### Expected Behavior
The composite control content within the grid row is hidden.

### Actual Behavior
The composite control content within the grid row is partially visible.

### Basic Information

- Version with issue: XF 4.2.0.848062
- Last known good version: XF 3.4.0.1008975
- IDE: Visual Studio 2019 Enterprise 
- Platform Target Frameworks:
  - Android: Target Framework 9
- Affected Devices:
  - Galaxy S8+ with Android 8.0.0

### Screenshots
![Normal Height](https://user-images.githubusercontent.com/8833988/67236354-bd0d5d80-f40e-11e9-94ca-dde6d6bee94e.jpg)

![Zero Height](https://user-images.githubusercontent.com/8833988/67236352-bd0d5d80-f40e-11e9-8e4f-ffce528c83cb.jpg)

![Zero Height other row](https://user-images.githubusercontent.com/8833988/67236434-e3cb9400-f40e-11e9-97ca-3a583121c22f.jpg)

### Reproduction Link

[GridRowHeightRedraw.zip](https://github.com/xamarin/Xamarin.Forms/files/3752545/GridRowHeightRedraw.zip)

**Comment by hartez** [comment]
@markdou812 Do you have any screenshots of what this should look like when it's behaving as you expect? I downgraded the project to 3.4.0.1008975 and ran it, but it looks the same as it does with 4.2.

**Comment by markdou812** [comment]
Apparently my sample solution didn't accurately reflect my production app issue. I feel like I can work around it for now but I'm am curious about whether or not this is the expected behavior - i.e. shouldn't collapsing the grid row hide the content in the row?

**Comment by BasileusBasileon** [comment]
@hartez I think I ran into this issue in a different context.
I was using a grid in a column of another grid. The width of that column was *. In the event that there is no space for this column, so its width is 0, the grid inside the column would correctly be set to a width of 0, but the view inside **that** grid would retain it's previous size.
I have uploaded a repo [here](https://github.com/BasileusBasileon/Xamarin-GridColumnWidth0) which reproduces the problem. Simply rotate an Android device from landscape to portrait, and it will occur.
The problem will also occur on UWP but is a bit more tricky to get to on account of the resizing being more gradual.
Interestingly, the view does retain its size on iOS, but is not shown over the other columns. 

To summarize, the behaviour I am expecting is that when a grids column gets set to a width of 0, and contains a grid with a view, then both that grid and its view get assigned a width of 0. Instead the view retains the size it had before.

**Comment by hartez** [comment]
This is definitely weird, though I'm not entirely sure it's a regression. It's certainly counterintuitive, though.

**Comment by demoh2019** [comment]
Update?
